powerfulseal/policy/action_alertmanager.py

```diff
@@ -0,0 +1,154 @@
+                    https=self.proxies.get("https", "")
+                ),
+                verify=False,
+            )
+            resp.raise_for_status()
+            success = True
+        except:
+            self.logger.exception("Exception while calling %s", url)
+        return success
+ 
+
+class ActionAlertManager(ActionAbstract):
+
+    def __init__(self, name, schema, logger=None, metric_collector=None):
+        self.name = name
+        self.schema = schema
+        self.logger = logger or makeLogger(__name__, name)
+        self.metric_collector = metric_collector or StdoutCollector()
+        self.targets = self.schema.get("targets", [])
+        self.proxies = self.schema.get("proxies", {})
+        self.silences = dict()
+        # build a url indexed map to store generated silences in mute action
+        for target in self.targets:
+            url = target.get("url", None)
+            if url is not None:
+                self.silences[url] = None
+        self.action_mapping = {
+            "mute": self.action_mute
+        }
+        self.cleanup_actions = []
+
+    def execute(self):
+        actions = self.schema.get("actions", [])
+        self.logger.info("executing alertmanager actions...")
+
+        success = True
+        for action in actions:
+            for key, method in self.action_mapping.items():
+                if key in action:
+                    params = action.get(key)
+                    ret = method(params)
+                    if not ret:
+                        success = False
+        return success
+
+    def action_mute(self, params):
+        success = True
+        # automute is True unless explicitly set to False
+        autoUnmute = params.get('autoUnmute', True) if type(params) is dict else True
+        for alert_manager in self.silences:
+            self.silences[alert_manager] = self.mute(alert_manager)
+            if self.silences[alert_manager] is not None:
+                if autoUnmute is True:
+                    # add unmute action for cleaning up when mute action generates silence
+                    # and `autoUnmute` is True
+                    self.cleanup_actions.append(
+                        ActionUnmuteAlertManager(
+                            name=self.name,
+                            silence_id=self.silences[alert_manager],
+                            alertmanager_url=alert_manager,
+                            proxies=self.proxies
+                    ))
+            elif success:
+                success = False
+        return success
+
+
+    def mute(self, url):
+        self.logger.info("Silencing alerts for alert manager %s", url)
+        startTime = datetime.now()
+        # mute all alerts for 15 mins (900 secs)
+        endTime = startTime + timedelta(seconds=900)
+        payload = {
+            'comment': 'silence all alerts',
+            'createdBy': 'powerfulseal',
+            'startsAt': startTime.isoformat(),
```

Comment:
I'm wondering if the timezones will ever be a problem since the API doesn't accept them.

Reply:
good point, i am going to use `astimezone()` to include OS timezone info. 
after doing that i can see:
```
2021-04-01 20:13:45 INFO action_alertmanager.Chaos itself Silenced alerts from 2021-04-01T20:13:45.508389+00:00 to 2021-04-01T20:23:45.508389+00:00
```
here `2021-04-01 20:03:19` is the OS local time (that node happens to use UTC)
and `2021-04-01T20:13:45.508389+00:00  ` is what we get from `startTime.isoformat()`
similarly `2021-04-01T20:23:45.508389+00:00` is what we get from `endTime.isoformate()`

Note that the payload sent to alert manager includes timezone offset (`+00:00` here) this should be enough for alertmanager to parse out correct time.